PROTOCOL.md

```diff
@@ -0,0 +1,193 @@
+# Twirp Wire Protocol
+
+This document defines the Twirp wire protocol over HTTP.
+
+## Conventions
+
+The requirement level keywords "MUST", "MUST NOT", "REQUIRED",
+"SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY",
+and "OPTIONAL" used in this document are to be interpreted as
+described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+The grammar rules used in this document are using [ABNF
+syntax](https://tools.ietf.org/html/rfc5234).
+
+## Overview
+
+The Twirp wire protocol is a simple RPC protocol based on HTTP and
+Protocol Buffers (proto). The protocol uses HTTP URLs to specify the
+RPC endpoints, and sends/receives proto messages as HTTP
+request/response bodies.
+
+To use Twirp, developers first define their APIs using proto files,
+then use Twirp tools to generate the client and the server libraries.
+The generated libraries implement the Twirp wire protocol, using the
+standard HTTP library provided by the programming language runtime or
+the operating system. Once the client and the server are implemented,
+the client can communicate with the server by making RPC calls.
+
+The Twirp wire protocol supports both binary and JSON encodings of
+proto messages, and works with any HTTP client and any HTTP version.
+However, certain capabilities may be limited by the actual HTTP
```

Comment:
I don't think it's worth including this caveat until we can actually say this precisely.